tests/unit/announcements.test.ts

```diff
@@ -0,0 +1,167 @@
+import { describe, expect, it, vi } from 'vitest';
+import {
+	Announcer,
+	PageAnnouncementOptions,
+	getPageAnnouncement
+} from '../../src/announcements.js';
+
+describe('announcer', () => {
+	describe('live region', () => {
+		it('creates an empty live region', () => {
+			const announcer = new Announcer();
+			const region = announcer.region;
+
+			expect(region).toBeInstanceOf(HTMLElement);
+			expect(region.id).toBe('swup-announcer');
+			expect(region.getAttribute('aria-live')).toBe('assertive');
+			expect(region.parentNode).toBe(document.body);
+			expect(region.textContent).toBe('');
+		});
+
+		it('creates an invisible live region', () => {
+			const announcer = new Announcer();
+			const region = announcer.region;
+
+			expect(region.style.position).toBe('absolute');
+			expect(region.style.top).toBe('0px');
+			expect(region.style.left).toBe('0px');
+		});
+
+		it('reuses existing live region', () => {
+			const existingRegion = document.createElement('div');
+			existingRegion.id = 'swup-announcer';
+			document.body.appendChild(existingRegion);
+
+			const announcer = new Announcer();
+			expect(announcer.region).toBe(existingRegion);
+		});
+	});
+
+	describe('announce', () => {
+		it('returns a promise', () => {
+			const announcer = new Announcer();
+			expect(announcer.announce('Hello')).toBeInstanceOf(Promise);
+		});
+
+		it('adds the announcement to the live region', async () => {
+			const announcer = new Announcer();
+			await announcer.announce('Hello');
+			expect(announcer.region.textContent).toBe('Hello');
+		});
+
+		it('modifies the announcement to allow duplicate messages', async () => {
+			const announcer = new Announcer();
+			const region = announcer.region;
+
+			await announcer.announce('Hello');
```

Comment:
That test description is a bit confusing to me. I suspect what it actually tests is that an unchanged announcement won't land as `${announcement}.` in the region? And that was a "hack" to make sure the region actually dispatches a change?

Reply:
I think "slightly modifies the announcement to allow announcing the same message twice in a row" would be the appropriate description 🤠 It's checking that it actually **will** land as `{message}.` in the region, which is the hack required to circumvent screen readers from ignoring technically unchanged live regions. Is the test description still confusing then?

Reply:
Or maybe "modifies each announcement to make it unique"?

Reply:
Aaah, now I understand! The announcer would switch back and forth between the message with and without the dot? That's amazingly clever (and also hard to understand in 3 months) 😃 

...I like both your suggestions for the test description better then the original. With a slight preference for the shorter one. Happy to let you choose :)

Reply:
Correct! It toggles the dot to allow that. I've used the shorter description for the test :)